internal/controller/nonadminbackup_controller.go

```diff
@@ -70,101 +67,180 @@ const (
-			return ctrl.Result{}, errUpdate
+			logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: New", nameField, nab.Name, constant.NameSpaceString, nab.Namespace)
+			return true, false, errUpdate
+		}
+
+		if updatedStatus {
+			logger.V(1).Info("NonAdminBackup CR - Requeue after Phase Update")
+			return false, true, nil
 		}
 	}
 
-	backupSpec, err := function.GetBackupSpecFromNonAdminBackup(&nab)
+	return false, false, nil
+}
+
+// ValidateVeleroBackupSpec validates the VeleroBackup Spec from the NonAdminBackup.
+//
+// Parameters:
+//
+//	ctx: Context for the request.
+//	logrLogger: Logger instance for logging messages.
+//	nab: Pointer to the NonAdminBackup object.
+//
+// The function attempts to get the BackupSpec from the NonAdminBackup object.
+// If an error occurs during this process, the function sets the NonAdminBackup status to "BackingOff"
+// and updates the corresponding condition accordingly.
+// If the BackupSpec is invalid, the function sets the NonAdminBackup condition to "InvalidBackupSpec".
+// If the BackupSpec is valid, the function sets the NonAdminBackup condition to "BackupAccepted".
+func (r *NonAdminBackupReconciler) ValidateVeleroBackupSpec(ctx context.Context, logrLogger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (exitReconcile bool, requeueReconcile bool, errorReconcile error) {
+	logger := logrLogger.WithValues("ValidateVeleroBackupSpec", nab.Namespace)
+
+	// Main Validation point for the VeleroBackup included in NonAdminBackup spec
+	_, err := function.GetBackupSpecFromNonAdminBackup(nab)
 
 	if err != nil {
 		errMsg := "NonAdminBackup CR does not contain valid BackupSpec"
 		logger.Error(err, errMsg)
 
-		// Phase: BackingOff
-		// BackupAccepted: False
-		// BackupQueued: False   # already set
-		updatedStatus, errUpdate := function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseBackingOff)
-		if errUpdate != nil {
-			logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: BackingOff", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-			return ctrl.Result{}, errUpdate
+		updatedStatus, errUpdateStatus := function.UpdateNonAdminPhase(ctx, r.Client, logger, nab, nacv1alpha1.NonAdminBackupPhaseBackingOff)
+		if errUpdateStatus != nil {
+			logger.Error(errUpdateStatus, "Unable to set NonAdminBackup Phase: BackingOff", nameField, nab.Name, constant.NameSpaceString, nab.Namespace)
+			return true, false, errUpdateStatus
 		} else if updatedStatus {
-			// We do not requeue as the State is BackingOff
-			return ctrl.Result{}, nil
+			// We do not requeue - the State was set to BackingOff
+			return true, false, nil
 		}
 
-		updatedStatus, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionFalse, "InvalidBackupSpec", errMsg)
-		if updatedStatus {
-			return ctrl.Result{}, nil
-		}
+		// Continue. VeleroBackup looks fine, setting Accepted condition
```

Comment:
I am assuming the NAB spec validation logic will reside here in between setting `BackupAccepted` as `False` and then setting `BackupAccepted` as `True`

Reply:
I think would be here https://github.com/mpryc/oadp-non-admin/blob/c1983f9f1ec3c603a8ad47f6deb7b0baf05ed66f/internal/common/function/function.go#L86

but that function validates and gets spec

I think would be better to split in 2 funtions

Reply:
yes splitting would make sense, I agree +1

Reply:
Can we have this as follow up PR? It's easier to test small change then such bigger one.

Reply:
yeah follow up would work, created an issue to track it https://github.com/migtools/oadp-non-admin/issues/60